src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -685,7 +685,7 @@ private static bool ShouldUseKnownLength(BoundCollectionExpression node, out int
             if (node is { Elements: [BoundCollectionExpressionSpreadElement { Expression: { } spreadExpression } spreadElement] }
                 && spreadElement.IteratorBody is BoundExpressionStatement expressionStatement)
             {
-                var spreadElementHasIdentityConversion = expressionStatement.Expression is not BoundConversion;
+                var spreadElementHasIdentityConversion = expressionStatement.Expression is not (BoundConversion or BoundKeyValuePairConversion);
```

Comment:
>BoundKeyValuePairConversion [](http://example.com/codeflow?start=115&length=27)

Does initial binding produce a `VisitAndRewriteCollectionElementExpression` when the element has KVP type and the collection element type is the same KVP?

Reply:
If there is an implicit conversion from source to destination `KeyValuePair<,>` type, initial binding should be generating a simple `BoundConversion` (or no conversion at all) rather than a `BoundKeyValuePairConversion`.